### PR TITLE
use withOrderMap for parameters

### PR DESCRIPTION
**Pull request by vhvb1989**
Using withOrder map for parameters instead of sorting keys

### CODE REVIEW

**Review by ellismg — COMMENTED** [review]
My personal preference is to keep the model types closer to their intended meaning, where key order does matter, and just do the sorting in places we need the stability.  I feel like the cognitive load of the ordered map here (and the slightly worse ergonomics in go) push us towards the current design.

But if we think this is likely to save us in other areas, then I'd be okay with us doing it.

Does seem strange that the MarshalJSON for the ordered map type which we are adding doesn't actually write the keys in the order they were added to the map.  The fact that this doesn't break end to ends is sort of proves that really order doesn't matter here.

File: cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go [hunk 1]
```diff
@@ -1952,23 +1957,21 @@ func (p *BicepProvider) ensureParameters(
...
+	parametersCount := len(parametersOrderedKeys)
+	if parametersCount == 0 {
 		return azure.ArmParameters{}, nil
 	}
-	configuredParameters := make(azure.ArmParameters, len(template.Parameters))
-
-	sortedKeys := maps.Keys(template.Parameters)
-	slices.Sort(sortedKeys)
-
+	configuredParameters := make(azure.ArmParameters, parametersCount)
```
Comment (weikanglim) [inline cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go:1965]: Can you provide more context here on the change? Is the old behavior that we'd read the parameters in an unordered fashion from the ARM JSON, and then impose our own ordering using sort? 

And we think the preservation of the ARM template JSON ordering is better?

**Review by wbreza — COMMENTED** [review]
Why does it matter the order of the parameters?  Are we solving a bug/issue?

**Comment by vhvb1989** [comment]
not required any more

**Pull request closed without merging**